sockeye/training.py

```diff
@@ -105,8 +105,8 @@ def _build_module(self, train_iter: data_io.BaseParallelSampleIter):
         num_factors = self.config.config_embed_source.num_factors
 
         source = mx.sym.Variable(C.SOURCE_NAME)
-        source_factors = [mx.sym.Variable('{}_factor_{:d}'.format(C.SOURCE_NAME, i)) for i in range(num_factors)]
-        source_length = utils.compute_lengths(source)
+        source_words = source.split(num_outputs=num_factors + 1, axis=2, squeeze_axis=True)[0]
```

Comment:
lets make this split optional to avoid the useless split for models without source factors.